### ISSUE TITLE
PEP8 suggestion of module-level dunders could lead to bug

### ISSUE THREAD

**Issue by clavedeluna** [python/cpython#101131]
Hi maintainers, please excuse if this isn't the right place to report this issue. I read [PEP1](https://peps.python.org/pep-0001/) but may be mistaken, please direct me to the right place and I can close and post elsewhere.

# Bug report

I'm reporting an issue documented in [this PEP section](https://peps.python.org/pep-0008/#module-level-dunder-names) which could lead to a bug.  I'm working on implementing this feature in [isort](https://github.com/PyCQA/isort/issues/1980) so your answer would help.

Basically, that PEP states all module-level dunder assignments should be at the top of the file but below `__future__`. That makes sense, except for this case:

```
from importlib import metadata

__version__ = metadata.version("isort")
```

In this example, if we were to move  the second line above the first line as described, you'd get a `NameError` as `metadata` is not defined.

**Comment by sobolevn** [comment]
Hi, I agree that the wording is not ideal. Can you please reopen this issue in https://github.com/python/peps ?

This is for bugs in CPython, not in PEPs :)
Thanks!
